From 42e662b11f4fe5d15ef59efb63eb224afb11572b Mon Sep 17 00:00:00 2001
From: Ricardo Lopez Esparza <ricardo.lopez.esparza@intel.com>
Date: Thu, 24 Mar 2016 23:50:22 -0600
Subject: [PATCH 10/10] ov8858: Fix power rails

Tunr v1p2 on instead of v1p5

Change-Id: I51dd2094aebc71b71d7b289555e38b077b703ff3
Original-Tracked_On: https://jira01.devtools.intel.com/browse/OAM-26493
Signed-off-by: Fernando Reyna <fernando.m.reyna.diaz@intel.com>
Reviewed-on: https://android.intel.com:443/481576
Signed-off-by: Lopez Esparza, Ricardo <ricardo.lopez.esparza@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/110166
Build: Dessert M BTNS (SC)
Approver: Mikael Persson <mikael.persson@intel.com>
Build: Dessert M CHT FFD (SC)
Build: Dessert M BXT RVP (SC)
Integrator: WeiX Meng <weix.meng@intel.com>
Reviewed-by: WeiX Meng <weix.meng@intel.com>
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-26488
Signed-off-by: Tao Wei <wei.tao@intel.com>
Reviewed-on: https://android.intel.com:443/487156
---
 .../camera/drivers/media/i2c/ov8858.c              | 27 ++++++----------------
 1 file changed, 7 insertions(+), 20 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/i2c/ov8858.c b/drivers/external_drivers/camera/drivers/media/i2c/ov8858.c
index 437f3c2..31832f2 100644
--- a/drivers/external_drivers/camera/drivers/media/i2c/ov8858.c
+++ b/drivers/external_drivers/camera/drivers/media/i2c/ov8858.c
@@ -750,26 +750,13 @@ static int __power_ctrl(struct v4l2_subdev *sd, bool flag)
 		return dev->platform_data->power_ctrl(sd, flag);
 
 #ifdef CONFIG_GMIN_INTEL_MID
-	/* CHT HR requires a power rail of 1.2v */
-	if (strcmp(dmi_get_system_info(DMI_BOARD_NAME), CHT_HR_DEV_NAME) == 0) {
-		if (dev->platform_data->v1p2_ctrl) {
-			ret = dev->platform_data->v1p2_ctrl(sd, flag);
-			if (ret) {
-				dev_err(&client->dev,
-						"failed to power %s 1.2v power rail\n",
-						flag ? "up" : "down");
-				return ret;
-			}
-		}
-	} else { /* CHT MRD requires a power rail of 1.6v */
-		if (dev->platform_data->v1p5_ctrl) {
-			ret = dev->platform_data->v1p5_ctrl(sd, flag);
-			if (ret) {
-				dev_err(&client->dev,
-						"failed to power %s 1.6v power rail\n",
-						flag ? "up" : "down");
-				return ret;
-			}
+	if (dev->platform_data->v1p2_ctrl) {
+		ret = dev->platform_data->v1p2_ctrl(sd, flag);
+		if (ret) {
+			dev_err(&client->dev,
+					"failed to power %s 1.2v power rail\n",
+					flag ? "up" : "down");
+			return ret;
 		}
 	}
 
-- 
1.9.1

